streamlit_app.py

```diff
@@ -0,0 +1,448 @@
+import streamlit as st
+
+import sys
+sys.path.append('src')
+import deck52
+import pprint
+import time
+import json
+import uuid
+import shelve
+
+import numpy as np
+
+import human
+import bots
+import sample
+
+from bidding import bidding
+from nn.models import Models
+from deck52 import decode_card
+from bidding.binary import DealData
+from objects import CardResp, Card
+
+# test dds load, encounter libboost_thread.so.1.71.0 not found
+#from ddsolver import dds
+
+def print(r):
+  st.write(r)
+
+#bids = st.text_input("Standard american biding convention", "1D 2D")
+#st.write(f"The meaning of {bids} is")
+
+
+def random_deal():
+    deal_str = deck52.random_deal()
+    auction_str = deck52.random_dealer_vuln()
+
+    return deal_str, auction_str
+
+
+class AsyncBotBid(bots.BotBid):
```

Comment:
since we are now using these async bots in several places, it might be better to factor them out into a separate module rather than copy-pasting them

Reply:
will check this weekend